setup/blocks/base_block.php

```diff
@@ -1,28 +1,53 @@
 <?php
 
-class BaseBlock {
-    public function __construct($block_name = '', $block_params = array(), $group_params = array(), $fields_params = array()){
-
-        if (empty($block_name) || empty($block_params) || empty($group_params) || empty($fields_params)) {
-            return;
-        }
+abstract class BaseBlock {
+    public function __construct(){
+        $this->validate_required_constants();
+        $this->register();
+    }
 
+    protected function register(){
         if( function_exists('create_acf_block') ) {
-            create_acf_block($block_name, $block_params);
+            create_acf_block($this->name(), $this->params());
         }
 
         if( function_exists('acf_add_local_field_group') ) {
-            acf_add_local_field_group($group_params);
+            acf_add_local_field_group($this->groupParams());
         }
 
         if( function_exists('acf_add_local_field') ) {
-            acf_add_local_field($fields_params);
+            acf_add_local_field($this->fieldsParams());
+        }
+    }
+
+    protected function validate_required_constants() {
+        $caller = get_called_class();
+        $required = ["$caller::NAME", "$caller::PARAMS", "$caller::FIELD_GROUP_PARAMS", "$caller::FIELDS_PARAMS"];
```

Comment:
Forse funzionerebbe anche usare `self::` per "visitare" le costanti in modo privato. non ci ho pensato di provare, lascio fare a te la prova se lo reputi curioso e/o più pulito